### ISSUE TITLE
phantomjs: post install error, TypeError: Path must be a string

### ISSUE THREAD

**Issue by youngboy** [Medium/phantomjs#535]
Trying to install phantomjs-prebuild@2.1.7 failed inside docker,

Here is error output

```
19-Apr-2016 14:09:52	[phantomjs-prebuilt@2.1.7] scripts.install: "node install.js" at /app/node_modules/.npminstall/phantomjs-prebuilt/2.1.7/phantomjs-prebuilt
19-Apr-2016 14:09:55	Considering PhantomJS found at /app/node_modules/.bin/phantomjs
19-Apr-2016 14:09:55	Looks like an `npm install -g`
19-Apr-2016 14:09:55	Error checking path, continuing { [Error: Cannot find module '/app/node_modules/.npminstall/phantomjs-prebuilt/2.1.7/phantomjs-prebuilt/lib/location'] code: 'MODULE_NOT_FOUND' }
19-Apr-2016 14:09:55	Phantom installation failed [TypeError: Path must be a string. Received undefined] TypeError: Path must be a string. Received undefined
19-Apr-2016 14:09:55	    at assertPath (path.js:8:11)
19-Apr-2016 14:09:55	    at Object.posix.join (path.js:479:5)
19-Apr-2016 14:09:55	    at findSuitableTempDirectory (/app/node_modules/.npminstall/phantomjs-prebuilt/2.1.7/phantomjs-prebuilt/install.js:127:30)
19-Apr-2016 14:09:55	    at /app/node_modules/.npminstall/phantomjs-prebuilt/2.1.7/phantomjs-prebuilt/install.js:476:19
19-Apr-2016 14:09:55	    at nextTickCallback (/app/node_modules/.npminstall/kew/0.7.0/kew/kew.js:47:28)
19-Apr-2016 14:09:55	    at nextTickCallbackWith0Args (node.js:420:9)
19-Apr-2016 14:09:55	    at process._tickCallback (node.js:349:13)
19-Apr-2016 14:09:55	Error: post install error, please remove node_modules before retry!
19-Apr-2016 14:09:55	Run "sh -c node install.js" error, exit code 1
```

I got the same `MODULE_NOT_FOUND` error in locally, while everything is fine.
so I think that is normal behavior, and look up install.js;

```
  var candidateTmpDirs = [
    process.env.TMPDIR || process.env.TEMP || process.env.npm_config_tmp,
    '/tmp',
    path.join(process.cwd(), 'tmp')
  ]

  for (var i = 0; i < candidateTmpDirs.length; i++) {
    var candidatePath = path.join(candidateTmpDirs[i], 'phantomjs')
  // ... 
```

so the actual reason is that all of my env `TMPDIR`, `TEMP`, `npm_config_tmp` are unset. and path.join will raise the error.

after setting `npm_config_tmp`, the npm install is success.

It could be nice to have defensive check on undefine value of candiataPath, or print some useful message to instruct developer what to do .

**Comment by nicks** [comment]
this should be fixed now

**Comment by ccorcos** [comment]
hmm fixed in what version? I just upgraded to 2.1.13 and I still have this issue.
